examples/cpp/hello_world/HelloWorld_main.cpp

```diff
@@ -0,0 +1,204 @@
+    CLIParser::hello_world_config config = CLIParser::parse_cli_options(argc, argv);
+
+    switch (config.entity)
+    {
+        case CLIParser::entity_kind::PUBLISHER:
+            entity_name = "Publisher";
+            samples = config.pub_config.samples;
+            try
+            {
+                publisher = new HelloWorldPublisher(config.pub_config, topic_name);
+                thread = new std::thread(&HelloWorldPublisher::run, publisher);
+            }
+            catch (const std::runtime_error& e)
+            {
+                EPROSIMA_LOG_ERROR(PUBLISHER, e.what());
+                ret = EXIT_FAILURE;
+            }
+            break;
+        case CLIParser::entity_kind::SUBSCRIBER:
+            samples = config.sub_config.samples;
+            if (config.sub_config.use_waitset)
+            {
+                entity_name = "Waitset Subscriber";
+                try
+                {
+                    subscriber_waitset = new HelloWorldSubscriberWaitset(config.sub_config, topic_name);
+                    thread = new std::thread(&HelloWorldSubscriberWaitset::run, subscriber_waitset);
+                }
+                catch (const std::runtime_error& e)
+                {
+                    EPROSIMA_LOG_ERROR(SUBSCRIBER, e.what());
+                    ret = EXIT_FAILURE;
+                }
+            }
+            else
+            {
+                entity_name = "Subscriber";
+                try
+                {
+                    subscriber = new HelloWorldSubscriber(config.sub_config, topic_name);
+                    thread = new std::thread(&HelloWorldSubscriber::run, subscriber);
+                }
+                catch (const std::runtime_error& e)
+                {
+                    EPROSIMA_LOG_ERROR(SUBSCRIBER_WAITSET, e.what());
+                    ret = EXIT_FAILURE;
+                }
+            }
+            break;
+        default:
+            EPROSIMA_LOG_ERROR(CLI_PARSER, "unknown entity");
+            CLIParser::print_help(EXIT_FAILURE);
+            break;
+    }
+
+    if (samples == 0)
+    {
+        std::cout << entity_name << " running. Please press Ctrl+C to stop the "
+                  << entity_name << " at any time." << std::endl;
+    }
+    else
+    {
+        switch (config.entity)
+        {
+            case CLIParser::entity_kind::PUBLISHER:
+                std::cout << entity_name << " running " << samples << " samples. Please press Ctrl+C to stop the "
+                          << entity_name << " at any time." << std::endl;
+                break;
+            case CLIParser::entity_kind::SUBSCRIBER:
+            default:
+                std::cout << entity_name << " running until " << samples << " samples have been received. Please press "
+                          << "Ctrl+C to stop the " << entity_name << " at any time." << std::endl;
+                break;
+        }
+    }
+
+    signal_handler = [&](std::string signal)
```

Comment:
I am keeping the `std:function` because handler registration fails:
`error: cannot convert ‘main(int, char**)::<lambda(int)>’ to ‘__sighandler_t’ {aka ‘void (*)(int)’}`

I've improved anyways its usage in [fbb9c0e](https://github.com/eProsima/Fast-DDS/pull/4547/commits/fbb9c0e5219cb41c9fdd6dc1df030640540adfb6)